infra/getting_started/your_first_lambda_function.md

```diff
@@ -0,0 +1,187 @@
+
+    The handler in this case is our function `hello`. This handler is relatively simple. It doesn't take any internal Lambda requests and just returns the classic "Hello, world!".
+
+7. All that's left is to transform our Go program into something Lambda can actually run. We can accomplish this by creating a binary and storing it in a `.zip` file.
+
+    ```
+    GOOS=linux GOARCH=amd64 go build -o main main.go
+    zip main.zip main
+    ```
+
+## Deploying with Terraform
+1. In the `hello-lambda` directory, create a new file called `main.tf`.
+2. Open that new file up in your favorite text editor or IDE.
+3. Before we create our Lambda function, we need to tell Terraform the basics such as that we'll be deploying to AWS and what region we'll be deploying this function to.
+
+    ```
+    variable "aws_region" {
+        default = "us-west-2"
+    }
+
+    provider "aws" {
+        region = "${var.aws_region}"
+    }
+    ```
+
+4. Let's create that Lambda function. Add the following lines to the file.
+
+    ```
+    resource "aws_lambda_function" "test_lambda" {
+        filename = "main.zip"
+        function_name = "test_lambda"
+        role             = "${aws_iam_role.iam_for_lambda_tf.arn}"
+        handler          = "main"
+        source_code_hash = "${base64sha256(file("main.zip"))}"
+        runtime          = "go1.x"
+    }
+    ```
+
+    `filename` points to where our `.zip` file is.
+
+    `function_name` is the name given to the function. We can invoke our Lambda function by calling this name.
+
+    `role` is the IAM role attached to this function. We will be creating this in the next step.
+
+    `handler` is the entrypoint into the function. In this particular case, it's the `main.go` file in `.zip`.
+
+    `runtime` specifies what language this function will be running in.
+
+    `source_code_hash` is used to trigger updates when your source code changes.
+
+5. What's left to do finish `main.tf` is creating that IAM role.
+    ```
+    resource "aws_iam_role" "iam_for_lambda_tf" {
+        name = "iam_for_lambda_tf"
+
+        assume_role_policy = <<EOF
+        {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+            "Action": "sts:AssumeRole",
+            "Principal": {
+                "Service": "lambda.amazonaws.com"
+            },
+            "Effect": "Allow",
+            "Sid": ""
+            }
+        ]
+        }
+    EOF
+    }
+    ```
+
+    This particular function doesn't need access to other AWS services so we're giving it a role with a no access policy.
+
+6. Save the file.
+7. Back on the command line, it's time to initialize the setup for your Terraform plan.
+   ```
+   terraform init
+   ```
+8. If that's looking good, let's check out what our plan looks like.
+    ```
+    terraform plan
+    ```
+9. Provided there's nothing unsurprising in the plan, it's time to deploy your function!
+    ```
+    terraform apply
```

Comment:
oooh - what does it do. Unfortunately I'm on a plane, otherwise I'd try it out,,,